cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -225,7 +225,7 @@ class concurrent_unordered_map {
         atomicCAS(&(insert_location->first), m_unused_key, insert_pair.first)};
 
     // Hash bucket empty
-    if (m_equal(m_unused_key, old_key)) {
+    if (m_unused_key == old_key) {
```

Comment:
atomicCAS only works correctly on int16, int32 and int64 types which can be directly compared.  If we're doing indirect insertion, insert_pair.first and old_key are going to be integer types that identify the index into the array.  We want to actually compare the indices rather than what the indices point to (since m_unused_key is most likely out of bounds on the indirect array).